### PR TITLE
add new page for teams API

### PR DESCRIPTION
**Pull request by fran-tirapu**
Task: https://trello.com/c/Q9FqpxCy/2037-documentation-for-user-management

New page for teams API added to documentation with the endpoints suggested.

Probably for our users it is also necessary to document `GET /:team_id` , to have direct access to "invitation status" and "user_ids" of the team.

### CODE REVIEW

File: content/rest-api/teams.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+        "email": "your_user@domain.com",
+        "role": "owner"
+     }' \
+     -X PUT https://api.codemagic.io/<team_id>/collaborator/<user_id>
+{{< /highlight >}}
+
+
+#### Response
+
+A collaborator objects is returned. You can see that the role is changed:
```
Comment (helinanever) [inline content/rest-api/teams.md:101]:
```suggestion
A collaborator object is returned. You can see that the role has changed:

```

File: content/rest-api/teams.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+            "addedBy":"<user_id>",
+            "activatedAt":"<date>"
+        }
+}
+
+{{< /highlight >}}
+
+
+
+## Delete a team
```
Comment (helinanever) [inline content/rest-api/teams.md:130]: Delete a team member?

File: content/rest-api/teams.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+{{< highlight bash "style=paraiso-dark">}}
+curl -H "Content-Type: application/json" \
+     -H "x-auth-token: <API Token>" \
+     -X DELETE https://api.codemagic.io/<team_id>/collaborator/<user_id>
+{{< /highlight >}}
+
+
+#### Response
+
+A full team object is returned. You can see that your team member  is removed
```
Comment (helinanever) [inline content/rest-api/teams.md:146]:
```suggestion
A full team object is returned. You can see that your team member has been removed

```

File: content/rest-api/teams.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+from `collaborators` section:
```
Comment (helinanever) [inline content/rest-api/teams.md:147]:
```suggestion
from the `collaborators` section: 

```

File: content/rest-api/teams.md [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+from `collaborators` section: 
+
+{{< highlight json "style=paraiso-dark">}}
+{
+  "team":{
+    "_id":"<team_id>",
+    "applicationIds":[...],
+    "billing":{...},
+    "buildTimes": {...},
+    "collaborators":[...],  // <--- your team member is removed
```
Comment (helinanever) [inline content/rest-api/teams.md:156]:
```suggestion
    "collaborators":[...],  // <--- your team member has been removed

```

**Review by helinanever — CHANGES_REQUESTED** [review]

File: content/rest-api/teams.md [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+Change the role of your team members. The available roles are `owner` and `developer` (corresponds to "member" in the UI).
+
+`PUT /:team_id/collaborator/:user_id`
+
+#### Parameters
+
+
+| **Name** | **Type** | **Description**                                            |
+|----------| -------- |------------------------------------------------------------|
+| `role`   | `string` | **Required.** Could be `owner` or `developer`              |
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:86]: is there a way to find `user id` from previous response?

File: content/rest-api/teams.md [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+#### Example
+
+{{< highlight bash "style=paraiso-dark">}}
+curl -H "Content-Type: application/json" \
+     -H "x-auth-token: <API Token>" \
+     -X DELETE https://api.codemagic.io/<team_id>/collaborator/<user_id>
+{{< /highlight >}}
+
+
+#### Response
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:150]: we don't need this information

File: content/rest-api/teams.md (hunk 6 above)
Comment (fran-tirapu) [reply content/rest-api/teams.md:86]: Yes, you can find it in `collaborators` section, but for me not make sense to have to create an invitation just to get this information. That's why I suggest in PR description to add the GET endpoint.

File: content/rest-api/teams.md [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+Invite a new team member to your team.
+
+`POST /:team_id/invitation`
+
+#### Parameters
+
+
+| **Name** | **Type** | **Description**                                            |
+|----------| -------- |------------------------------------------------------------|
+| `email`  | `string` | **Required.** SSH or HTTPS URL for cloning the repository. |
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:25]: description is copy-paste

File: content/rest-api/teams.md [hunk 9]
```diff
@@ -0,0 +1,165 @@
...
+ where you can locate their `user_ids`.
+
+
+{{< highlight json "style=paraiso-dark">}}
+{
+  "team":{
+    "_id":"<team_id>",
+    "applicationIds":[...],
+    "billing":{...},
+    "buildTimes": {...},
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:55]: let's keep only `invitations` and I don't see if we return `id` in invitations

File: content/rest-api/teams.md [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+
+#### Example
+
+How to upgrade a current member of your team to team owner:
+
+{{< highlight bash "style=paraiso-dark">}}
+curl -H "Content-Type: application/json" \
+     -H "x-auth-token: <API Token>" \
+     -d '{
+        "email": "your_user@domain.com",
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:114]: is `email` required here or it's copy-paste?

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/rest-api/teams.md (hunk 9 above)
Comment (fran-tirapu) [reply content/rest-api/teams.md:55]: No, I think user is not created already, only the invitation.

Comment (mikhail-tokarev) [reply content/rest-api/teams.md:55]: ah I see. Can you test if invitation endpoint works correctly with `owner` role?
if it works as described, we can completely remove `change a team member role` section and this example. Sorry for the late idea

**Review by mikhail-tokarev — CHANGES_REQUESTED** [review]

File: content/rest-api/teams.md (hunk 9 above)
Comment (fran-tirapu) [reply content/rest-api/teams.md:55]: yes, you can invite a member as developer or as owner and work as expected

File: content/rest-api/teams.md [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+            "createdAt": "<create_date>",
+            "email": "your_user@domain.com",
+            "permissions": [...],
+            "inviteCode": "<invite_code>"}
+        ],
+        ...
+        ...
+    }
+}
+{{< /highlight >}}
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:88]:
```suggestion
```

File: content/rest-api/teams.md [hunk 12]
```diff
@@ -0,0 +1,103 @@
...
+     -X POST https://api.codemagic.io/<team_id>/invitation
+{{< /highlight >}}
+
+
+#### Response
+
+A full team object is returned. You can see that your invitation has been created
+in the `invitations` section.
+Also, you can see your current active team members in the `collaborators` section,
+ where you can locate their `user_ids`.
```
Comment (mikhail-tokarev) [inline content/rest-api/teams.md:46]:
```suggestion
A full team object is returned.
```

**Review by mikhail-tokarev — APPROVED** [review]

File: content/rest-api/teams.md [hunk 13]
```diff
@@ -0,0 +1,60 @@
...
+
+`POST /:team_id/invitation`
+
+#### Parameters
+
+
+| **Name** | **Type** | **Description**                               |
+|----------| -------- |-----------------------------------------------|
+| `email`  | `string` | **Required.** User email                    |
+| `role`   | `string` | **Required.** Could be `owner` or `developer` |
```
Comment (helinanever) [inline content/rest-api/teams.md:26]: I would still like to see a note about the developer role corresponding to the member role as seen in the UI.

**Review by helinanever — COMMENTED** [review]
Left a note about user roles.
Also, I think the file could use some cleanup -- there are lots of unnecessary empty rows.

File: content/rest-api/teams.md [hunk 14]
```diff
@@ -0,0 +1,55 @@
...
+     }' \
+     -X POST https://api.codemagic.io/<team_id>/invitation
+{{< /highlight >}}
+
+#### Response
+
+A full team object is returned.
+
+## Delete a team member
+
```
Comment (helinanever) [inline content/rest-api/teams.md:45]: Inviting a team member has body text, so would be nice if this section had one, too. 

```suggestion
Remove a team member from the team.
```

**Review by helinanever — APPROVED** [review]
LGTM

**Pull request merged**